### PR TITLE
[guides] add a section for ordering with a nested association attribute

### PR DESCRIPTION
**Pull request by JulienAmoros**
As we were working with ROM, we had similar use case and found a trick to solve this, we decided to share this back.

What do you think?

### CODE REVIEW

File: source/4.0/guides/sql-how-to/order-by-nested-association-attribute.html.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+but we want them ordered by Team name, and if they're in the same team, we want them to be 
+ordered by creation_date.
+
+``` ruby
+module Repositories
+  class Users
+    def get_ordered_by_team_and_creation_date
+      users
+        .combine(:team)
+        .order('team[:name]', :created_at)
```
Comment (solnic) [inline source/4.0/guides/sql-how-to/order-by-nested-association-attribute.html.md]: This does not look like a valid syntax. You would have to join teams to be able to order by one of the columns from team table.

Comment (JulienAmoros) [reply source/4.0/guides/sql-how-to/order-by-nested-association-attribute.html.md]: Oh yes, you're right, there's something messed up here 🤔 
I'm fixing this!

**Comment by JulienAmoros** [comment]
@solnic it's fixed, also I allowed myself to copy this in new 5.0 folder, I'll do the same with my other PRs if you validate them and if things mentionned still work in 5.0.

File: source/4.0/guides/sql-how-to/order-by-nested-association-attribute.html.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+but we want them ordered by Team name, and if they're in the same team, we want them to be 
+ordered by creation date.
+
+``` ruby
+module Repositories
+  class Users
+    def get_ordered_by_team_and_creation_date
+      users
+        .join(:teams)
+        .order { |r| [r[:teams][:name], r[:users][:created_at]] }
```
Comment (solnic) [inline source/4.0/guides/sql-how-to/order-by-nested-association-attribute.html.md]: `order(teams[:name], teams[:created_at])` would be simpler

File: source/4.0/guides/sql-how-to/order-by-nested-association-attribute.html.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    end
+  end
+end
+
+Repositories::Users.get_ordered_by_team_and_creation_date
+```
+
+If you want to know more about how you can use "#order" method, you can take a look at the 
+['rom-sql' specs](https://github.com/rom-rb/rom-sql/blob/73701c35656501045b52859671e4acf5fab35905/spec/unit/relation/order_spec.rb)
+or in [Sequel documentation](https://www.rubydoc.info/gems/sequel/4.38.0/Sequel%2FDataset:order).
```
Comment (solnic) [inline source/4.0/guides/sql-how-to/order-by-nested-association-attribute.html.md]: Please just link to our [API docs](https://api.rom-rb.org/rom-sql/ROM/SQL/Relation/Reading.html#order-instance_method) instead. Pointing to Sequel docs may only add confusion, Sequel is *an implementation detail* in rom-sql, and even though some methods are similar, they are not the same thing as relation methods.

File: source/4.0/guides/sql-how-to/order-by-nested-association-attribute.html.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+      attribute :name, ROM::Types::String
+
+      associations do
+        has_many :users
+      end
+    end
+  end
+end
+
+Repositories::Users.get_ordered_by_team_and_creation_date
```
Comment (solnic) [inline source/4.0/guides/sql-how-to/order-by-nested-association-attribute.html.md]: I wonder if it makes sense to use a repository in this guide. How about just using a relation and showing how the SQL would look like?

File: source/4.0/guides/sql-how-to/order-by-nested-association-attribute.html.md (hunk 2 above)
Comment (JulienAmoros) [reply source/4.0/guides/sql-how-to/order-by-nested-association-attribute.html.md]: I don't understand why I get that:
```
2.5.3 :007 > user_repo.users.left_join(:teams).order(teams[:name])
Traceback (most recent call last):
        2: from ./app.rb:5:in `<main>'
        1: from (irb):7
NameError (undefined local variable or method `teams' for main:Object)
```
Here is my Relations' code:
```
module Relations
  class Users < ROM::Relation[:sql]
    schema(:users) do
      attribute :id, ROM::Types::Integer
      attribute :name, ROM::Types::String
      attribute :created_at, ROM::Types::DateTime
      attribute :team_id, ROM::Types::Integer
      primary_key :id

      associations do
        belongs_to :team
      end
    end
  end

  class Teams < ROM::Relation[:sql]
    schema(:teams) do
      attribute :id, ROM::Types::Integer
      attribute :name, ROM::Types::String
      primary_key :id

      associations do
        has_many :users
      end
    end
  end
end
```
Is there anything I missed?
Plus, I want to order by the `name` of the team, and then by `created_at` attribute of my user

File: source/4.0/guides/sql-how-to/order-by-nested-association-attribute.html.md (hunk 4 above)
Comment (JulienAmoros) [reply source/4.0/guides/sql-how-to/order-by-nested-association-attribute.html.md]: You're right, that would be better!

File: source/4.0/guides/sql-how-to/order-by-nested-association-attribute.html.md (hunk 2 above)
Comment (solnic) [reply source/4.0/guides/sql-how-to/order-by-nested-association-attribute.html.md]: You tried to access a relation in the context of the main object, so it crashed. What you probably meant to do was:

```ruby
user_repo.users.left_join(:teams).order(user_repo.teams[:name])
```

...but as I said, no need to use repos in this guide.

File: source/5.0/guides/sql-how-to/order-by-nested-association-attribute.html.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+ INNER JOIN `teams` ON (`users`.`team_id` = `teams`.`id`) ORDER BY `teams`.`name`, `users`.
+ `created_at`">>
+2.5.3 :004 > users.combine(:teams).join(:teams).order { |r| [r[:teams][:name], r[:users][:created_at]] }.to_a
+ => [{:id=>1, :name=>"John Doe", :created_at=>2019-07-07 15:54:14 +0200, :team_id=>1, :team=>{:id=>1, :name=>"A team"}},
+ {:id=>3, :name=>"Jack Doe", :created_at=>2019-07-07 15:54:27 +0200, :team_id=>1, :team=>{:id=>1, :name=>"A team"}},
+ {:id=>2, :name=>"Jane Doe", :created_at=>2019-07-07 15:54:22 +0200, :team_id=>2, :team=>{:id=>2, :name=>"B team"}}]
+```
+
+If you want to know more about how you can use "#order" method, you can take a look at the 
+['rom-sql' specs](https://github.com/rom-rb/rom-sql/blob/73701c35656501045b52859671e4acf5fab35905/spec/unit/relation/order_spec.rb).
```
Comment (solnic) [inline source/5.0/guides/sql-how-to/order-by-nested-association-attribute.html.md]: Please *don't point to specs*, instead, it'd better to link to API I docs.

File: source/4.0/guides/sql-how-to/order-by-nested-association-attribute.html.md (hunk 2 above)
Comment (JulienAmoros) [reply source/4.0/guides/sql-how-to/order-by-nested-association-attribute.html.md]: Oh ok! I got it now! Thanks!

**Comment by JulienAmoros** [comment]
Sorry @solnic I misread your comment the first time, anyway thank you so much for your feedbacks.
If you merge this PR I'll update the other ones accordingly (I'll probably have conflicts on the structure)

**Comment by solnic** [comment]
Merged. Thank you 😄

**Pull request merged**